typescript/deploy-sdk/src/AltVMCoreModule.ts

```diff
@@ -318,27 +326,78 @@ export class AltVMCoreModule implements HypModule<CoreModuleType> {
     deployedIsm: Address;
     ismUpdateTxs: AnnotatedTx[];
   }> {
-    const { mailbox } = this.serialize();
+    // If expected ISM is an address reference, use it directly
+    if (typeof expectDefaultIsmConfig === 'string') {
+      return {
+        deployedIsm: expectDefaultIsmConfig,
+        ismUpdateTxs: [],
+      };
+    }
 
-    const ismModule = new AltVMIsmModule(
+    const chainMetadata = this.chainLookup.getChainMetadata(this.args.chain);
+    const writer = createIsmWriter(
+      chainMetadata,
       this.chainLookup,
-      {
-        addresses: {
-          mailbox: mailbox,
-          deployedIsm: actualDefaultIsmConfig.address,
-        },
-        chain: this.chainName,
-        config: actualDefaultIsmConfig.address,
-      },
       this.signer,
     );
+
+    // Read actual ISM state
+    const actualArtifact = await writer.read(actualDefaultIsmConfig.address);
+
+    // Convert expected config to artifact format
+    const expectedArtifact = ismConfigToArtifact(
+      expectDefaultIsmConfig,
+      this.chainLookup,
+    );
+
     this.logger.info(
       `Comparing target ISM config with ${this.args.chain} chain`,
     );
-    const ismUpdateTxs = await ismModule.update(expectDefaultIsmConfig);
-    const { deployedIsm } = ismModule.serialize();
 
-    return { deployedIsm, ismUpdateTxs };
+    // Decide: deploy new ISM or update existing one
+    if (
+      this.shouldDeployNewIsm(actualArtifact.config, expectedArtifact.config)
+    ) {
+      // Deploy new ISM
+      const [deployed] = await writer.create(expectedArtifact);
+      return {
+        deployedIsm: deployed.deployed.address,
+        ismUpdateTxs: [],
+      };
+    }
+
+    // Update existing ISM (only routing ISMs support updates)
+    const deployedArtifact: DeployedIsmArtifact = {
+      ...expectedArtifact,
+      artifactState: ArtifactState.DEPLOYED,
+      config: expectedArtifact.config as IsmArtifactConfig,
```

Comment:
I don't thin we need this type assertion here as at this point the compiler should be clever enough to understand that this is the routing ism config

<img width="673" height="367" alt="Image" src="https://github.com/user-attachments/assets/567f841f-09e5-4fb9-abe6-6621c650f1cf" />